From eb3ffa88e42c7fcfe71d593339dac7e73c75bba2 Mon Sep 17 00:00:00 2001
From: Carlos Garnacho <carlosg@gnome.org>
Date: Wed, 26 Aug 2020 23:56:45 +0200
Subject: [PATCH] schemas: Move antialiasing/hinting/rgba-order settings here

These settings reside in gnome-settings-daemon, but are actually accessed
all over the place. It seems to make more sense to have them here, even
though they still would be consumed by gnome-settings-daemon (x11 only,
mind you).

One of these consumers is GTK (wayland backend), possibly via the settings
portal, so it's nice to have these settings exist without gnome-settings-daemon
installed.

Fixes: https://gitlab.gnome.org/GNOME/gsettings-desktop-schemas/-/issues/31
---
 headers/gdesktop-enums.h                      | 24 +++++++++++++++++++
 ...org.gnome.desktop.interface.gschema.xml.in | 21 ++++++++++++++++
 2 files changed, 45 insertions(+)

diff --git a/headers/gdesktop-enums.h b/headers/gdesktop-enums.h
index 2ab6643..dac464b 100644
--- a/headers/gdesktop-enums.h
+++ b/headers/gdesktop-enums.h
@@ -227,4 +227,28 @@ typedef enum
   G_DESKTOP_USB_PROTECTION_ALWAYS
 } GDesktopUsbProtection;
 
+typedef enum
+{
+  G_DESKTOP_FONT_ANTIALIASING_MODE_NONE,
+  G_DESKTOP_FONT_ANTIALIASING_MODE_GRAYSCALE,
+  G_DESKTOP_FONT_ANTIALIASING_MODE_RGBA,
+} GDesktopFontAntialiasingMode;
+
+typedef enum
+{
+  G_DESKTOP_FONT_HINTING_NONE,
+  G_DESKTOP_FONT_HINTING_SLIGHT,
+  G_DESKTOP_FONT_HINTING_MEDIUM,
+  G_DESKTOP_FONT_HINTING_FULL,
+} GDesktopFontHinting;
+
+typedef enum
+{
+  G_DESKTOP_FONT_RGBA_ORDER_RGBA,
+  G_DESKTOP_FONT_RGBA_ORDER_RGB,
+  G_DESKTOP_FONT_RGBA_ORDER_BGR,
+  G_DESKTOP_FONT_RGBA_ORDER_VRGB,
+  G_DESKTOP_FONT_RGBA_ORDER_VBGR,
+} GDesktopFontRgbaOrder;
+
 #endif /* __gdesktop_enums_h__ */
diff --git a/schemas/org.gnome.desktop.interface.gschema.xml.in b/schemas/org.gnome.desktop.interface.gschema.xml.in
index e395325..2e3554a 100644
--- a/schemas/org.gnome.desktop.interface.gschema.xml.in
+++ b/schemas/org.gnome.desktop.interface.gschema.xml.in
@@ -273,5 +273,26 @@
         If true, pressing a key will highlight the current pointer location on screen.
       </description>
     </key>
+    <key name="font-antialiasing" enum="org.gnome.desktop.GDesktopFontAntialiasingMode">
+      <default>'grayscale'</default>
+      <summary>Antialiasing</summary>
+      <description>
+        The type of antialiasing to use when rendering fonts. Possible values are: “none” for no antialiasing, “grayscale” for standard grayscale antialiasing, and “rgba” for subpixel antialiasing (LCD screens only).
+      </description>
+    </key>
+    <key name="font-hinting" enum="org.gnome.desktop.GDesktopFontHinting">
+      <default>'slight'</default>
+      <summary>Hinting</summary>
+      <description>
+	The type of hinting to use when rendering fonts. Possible values are: “none” for no hinting and “slight” for fitting only to the Y-axis like Microsoft’s ClearType, DirectWrite and Adobe’s proprietary font rendering engine. Ignores native hinting within the font, generates hints algorithmically. Used on Ubuntu by default. Recommended. The meaning of “medium” and “full” depends on the font format (.ttf, .otf, .pfa/.pfb) and the installed version of FreeType. They usually try to fit glyphs to both the X and the Y axis (except for .otf: Y-only). This can lead to distortion and/or inconsistent rendering depending on the quality of the font, the font format and the state of FreeType’s font engines.
+      </description>
+    </key>
+    <key name="font-rgba-order" enum="org.gnome.desktop.GDesktopFontRgbaOrder">
+      <default>'rgb'</default>
+      <summary>RGBA order</summary>
+      <description>
+	The order of subpixel elements on an LCD screen; only used when antialiasing is set to “rgba”. Possible values are: “rgb” for red on left (most common), “bgr” for blue on left, “vrgb” for red on top, “vbgr” for red on bottom.
+      </description>
+    </key>
   </schema>
 </schemalist>
-- 
GitLab

